### PR TITLE
test: Remove some {DP,Services} Ginkgo test cases

### PR DESCRIPTION
**Pull request by brb**
See commit msgs.

### CODE REVIEW

File: test/k8s/datapath_configuration.go [hunk 1]
```diff
@@ -168,38 +151,6 @@ var _ = Describe("K8sDatapathConfig", func() {
...
-				"endpointRoutes.enabled": "true",
-			}
-			enableVXLANTunneling(options)
-			deploymentManager.DeployCilium(options, DeployCiliumOptionsAndDNS)
-			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
-
-			if helpers.RunsOn419OrLaterKernel() {
-				By("Test BPF masquerade")
-				Expect(testPodHTTPToOutside(kubectl, "http://google.com", false, false, false)).
-					Should(BeTrue(), "Connectivity test to http://google.com failed")
```
Comment (pchaigno) [inline test/k8s/datapath_configuration.go:199]: This is not covered by configuration 4 of ci-datapath. Do we know why BPF masquerading in the context of VXLAN+per endpoint routes specifically would matter?

File: test/k8s/datapath_configuration.go [hunk 2]
```diff
@@ -186,38 +186,6 @@ var _ = Describe("K8sDatapathConfig", func() {
...
-			deploymentManager.DeployCilium(map[string]string{
-				"tunnel":               "disabled",
-				"autoDirectNodeRoutes": "true",
-			}, DeployCiliumOptionsAndDNS)
-
-			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
-			if helpers.RunsOn419OrLaterKernel() {
-				By("Test BPF masquerade")
-				Expect(testPodHTTPToOutside(kubectl, "http://google.com", false, false, false)).
-					Should(BeTrue(), "Connectivity test to http://google.com failed")
```
Comment (pchaigno) [inline test/k8s/datapath_configuration.go:276]: Ditto.

**Review by pchaigno — APPROVED** [review]
LGTM with one potential small concern over two subtests.

File: test/k8s/datapath_configuration.go (hunk 1 above)
Comment (brb) [reply test/k8s/datapath_configuration.go:199]: Good catch, but it should be covered after https://github.com/cilium/cilium/pull/24210/files has been merged.

There is nothing specific about different routing modes when it comes to BPF masq, as it's entirely handled by `bpf_host`.

Comment (pchaigno) [reply test/k8s/datapath_configuration.go:199]: > There is nothing specific about different routing modes when it comes to BPF masq, as it's entirely handled by `bpf_host`.

I'm seeing a lot of macros on `TUNNEL_MODE` and `IS_BPF_OVERLAY` in the SNAT logic :grimacing: 
https://github.com/cilium/cilium/blob/853ec101d8adee8d482040aa038e5b71ce7775e2/bpf/lib/nat.h#L576

Comment (brb) [reply test/k8s/datapath_configuration.go:199]: Completely forgot about those. Thanks.

Anyway, #23210 was merged, so the BPF masq should be tested by ci-datapath.

Comment (pchaigno) [reply test/k8s/datapath_configuration.go:199]: Yep, completely covered by case 4 now.

File: test/k8s/datapath_configuration.go (hunk 2 above)
Comment (pchaigno) [reply test/k8s/datapath_configuration.go:276]: This is now covered by case 7 of ci-datapath.

**Review by YutaroHayakawa — APPROVED** [review]

**Review by christarazi — APPROVED** [review]

**Pull request merged**